src/main/java/net/buildtheearth/buildteam/components/generator/field/FieldScripts.java

```diff
@@ -0,0 +1,499 @@
+                operations++;
+
+                commands.add("//fast");
+
+            } else {
+                commands.add("//replace 35:5 60,3,5:1");
+                operations++;
+
+                commands.add("//fast");
+                commands.add("//replace >60,3,5:1 175");
+                operations++;
+
+                commands.add("//replace >175 175");
+                operations++;
+
+                commands.add("//fast");
+
+            }
+        }
+
+        if (crop == Crop.WHEAT) {
+            if (type == CropStage.LIGHT) {
+                commands.add("//replace 35:5 3,3:1");
+                operations++;
+
+                commands.add("//replace >3,3:1 107:4,107:5,107:6,107:7,184:4,184:5,184:6,184:7");
+                operations++;
+
+            } else {
+                commands.add("//replace 35:5 3,3:1,5,5:3");
+                operations++;
+
+                commands.add("//fast");
+
+                commands.add("//replace >3,3:1 31:1,175");
+                operations++;
+
+                commands.add("//fast");
+            }
+        }
+
+        if (crop == Crop.CATTLE || crop == Crop.MEADOW) {
+            commands.add("//gmask 0");
+
+            operations += Generator.createPolyLine(commands, points, "41", true, blocks, 1);
+            commands.add("//gmask");
+
+            Generator.createPolySelection(commands, points);
+
+            List<Vector> oneMeterPoints = new ArrayList<>(points);
+            oneMeterPoints = Generator.populatePoints(oneMeterPoints, 1);
+            List<Vector> fencePoints = new ArrayList<>(oneMeterPoints);
+            fencePoints = Generator.reducePoints(fencePoints, 3 + 1, 3 - 1);
+
+            commands.add("//sel cuboid");
+            commands.add("//expand 10 10 west");
+            commands.add("//expand 10 10 north");
+
+            for (Vector vector : fencePoints) {
+                commands.add("//pos1 " + Generator.getXYZWithVerticalOffset(vector, blocks, 1));
+                commands.add("//pos2 " + Generator.getXYZWithVerticalOffset(vector, blocks, 1));
+                commands.add("//set " + fence);
+            }
+
+            Generator.createPolySelection(commands, points);
+
+            commands.add("//sel cuboid");
+            commands.add("//gmask");
+
+            commands.add("//expand 10 10 up");
+            commands.add("//expand 10 10 west");
+            commands.add("//expand 10 10 north");
+
+            commands.add("//replace >41 77:5");
+            operations++;
+            commands.add("//replace 41 166");
+            operations++;
+
+            commands.add("//gmask !" + fence + ",77,166");
+            commands.add("//replace >35:5 70%0,30%31:1");
+            operations++;
+
+            if (crop == Crop.CATTLE) commands.add("//replace 35:5 60%3,20%2,20%3:1");
+            if (crop == Crop.MEADOW) commands.add("//replace 35:5 70%2,20%3,10%3:1");
+            operations++;
+
+        }
+
```

Comment:
Please restore the original selection once you are done generating. Take a look at the RoadScripts.java for reference